controllers/sync/policy_spec_sync.go

```diff
@@ -7,110 +7,59 @@ import (
-	}
-	var kubeClient kubernetes.Interface = kubernetes.NewForConfigOrDie(managedCfg)
-	eventsScheme := runtime.NewScheme()
-	if err = v1.AddToScheme(eventsScheme); err != nil {
-		return err
-	}
-
-	eventBroadcaster := record.NewBroadcaster()
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		log.Error(err, "Failed to get watch namespace")
-		return err
-	}
-	eventBroadcaster.StartRecordingToSink(&corev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events(namespace)})
-	managedRecorder := eventBroadcaster.NewRecorder(eventsScheme, v1.EventSource{Component: controllerName})
-
-	return add(mgr, newReconciler(mgr, managedClient, managedRecorder))
-}
-
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager, managedClient client.Client,
-	managedRecorder record.EventRecorder) reconcile.Reconciler {
-	return &ReconcilePolicy{hubClient: mgr.GetClient(), managedClient: managedClient,
-		managedRecorder: managedRecorder, scheme: mgr.GetScheme()}
-}
-
-// add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
-	// Create a new controller
-	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
-	}
-
-	// Watch for changes to primary resource Policy
-	err = c.Watch(&source.Kind{Type: &policiesv1.Policy{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		return err
-	}
-
-	return nil
+// SetupWithManager sets up the controller with the Manager.
+func (r *PolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&policiesv1.Policy{}).
+		Complete(r)
 }
 
 // blank assignment to verify that ReconcilePolicy implements reconcile.Reconciler
-var _ reconcile.Reconciler = &ReconcilePolicy{}
+var _ reconcile.Reconciler = &PolicyReconciler{}
 
 // ReconcilePolicy reconciles a Policy object
-type ReconcilePolicy struct {
+type PolicyReconciler struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
-	hubClient       client.Client
-	managedClient   client.Client
-	managedRecorder record.EventRecorder
-	scheme          *runtime.Scheme
+	HubClient       client.Client
+	ManagedClient   client.Client
+	ManagedRecorder record.EventRecorder
+	Scheme          *runtime.Scheme
 }
 
 // Reconcile reads that state of the cluster for a Policy object and makes changes based on the state read
 // and what is in the Policy.Spec
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
```

Comment:
I don't see any kubebuilder RBAC markers in this file (i.e. `//+kubebuilder:rbac`). Could you please add them to match what was originally in `deploy/operator.yaml`? It seems the `deploy/rbac/role.yaml` doesn't match what was there previously.